iree/hal/drivers/CMakeLists.txt

```diff
@@ -28,6 +28,11 @@ if(${IREE_HAL_DRIVER_VULKAN})
   list(APPEND IREE_HAL_DRIVER_MODULES iree::hal::vulkan::registration)
 endif()
 
+if(APPLE)
+  find_library(Metal Metal)
+  find_library(Foundation Foundation)
+endif()
```

Comment:
Can this go in https://github.com/google/iree/blob/main/iree/hal/metal/CMakeLists.txt and/or https://github.com/google/iree/blob/main/iree/hal/metal/registration/CMakeLists.txt instead?